drivers/SmartThings/philips-hue/src/handlers.lua

```diff
@@ -166,6 +166,28 @@ local function do_color_action(driver, device, args)
   end
 end
 
+-- Function to allow changes to "setHue" attribute to Philips Hue light devices
+---@param driver HueDriver
+---@param device HueChildDevice
+local function do_setHue_action(driver, device, args)
+
+  -- Use existing 'saturation' value for device or set to 0 and pass arg values to function 'do_color_action'
+  local currentSaturation = device:get_latest_state("main", capabilities.colorControl.ID, capabilities.colorControl.saturation.NAME) or 0
+  args.args.color.saturation = currentSaturation
+  do_color_action(driver, device, args)
+end
+
+-- Function to allow changes to "setSaturation" attribute to Philips Hue light devices
+---@param driver HueDriver
+---@param device HueChildDevice
+local function do_setSaturation_action(driver, device, args)
+
+  -- Use existing 'hue' value for device or set to 0 and pass arg values to function 'do_color_action'
+  local currentHue = device:get_latest_state("main", capabilities.colorControl.ID, capabilities.colorControl.hue.NAME) or 0
+  args.args.color.hue = currentHue
+  do_color_action(driver, device, args)
+end
+
```

Comment:
I let you astray here, by assuming that `setHue` and `setSaturation` used the same args payload as `setColor` but with a single field for either hue or sat. But it turns out that instead of `args.args.color.hue` and `args.args.color.saturation` it just uses `args.args.hue` or `saturation`

We can still implement this in a DRY way, though, by adding a `color` table to the args.

```suggestion
-- Function to allow changes to "setHue" attribute to Philips Hue light devices
---@param driver HueDriver
---@param device HueChildDevice
local function do_setHue_action(driver, device, args)

  -- Use existing 'saturation' value for device or set to 0 and pass arg values to function 'do_color_action'
  local currentSaturation = device:get_latest_state("main", capabilities.colorControl.ID, capabilities.colorControl.saturation.NAME) or 0
  args.args.color = {
    hue = args.args.hue,
    saturation = currentSaturation
  }
  do_color_action(driver, device, args)
end

-- Function to allow changes to "setSaturation" attribute to Philips Hue light devices
---@param driver HueDriver
---@param device HueChildDevice
local function do_setSaturation_action(driver, device, args)

  -- Use existing 'hue' value for device or set to 0 and pass arg values to function 'do_color_action'
  local currentHue = device:get_latest_state("main", capabilities.colorControl.ID, capabilities.colorControl.hue.NAME) or 0
  args.args.color = {
   hue = currentHue,
   saturation = args.args.saturation
  }
  do_color_action(driver, device, args)
end

```

Reply:
Thanks for catching that. I made the proposed change.